rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,899 @@
+                    The only side effect is that the property is not applied.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+            <where param="constant">
+                <matches pattern="(MAX_THREAD_STATES|TERM_INDEX_INTERVAL)" />
+            </where>
+        </rule>
+        <rule id="hsearch-00111">
+            <when>
+                <javaclass references="org.hibernate.search.annotations.Spatial" >
+                    <location>ANNOTATION</location>
+                </javaclass>
+                <!-- https://issues.jboss.org/browse/WINDUP-974 -->
+                <!-- This doesn't find annotation with enum element specified -->
+<!--                 <javaclass references="org.hibernate.search.annotations.SpatialMode.GRID" />                 -->
+            </when>
+            <perform>
+                <hint title="Renamed SpatialMode.GRID to HASH in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>The enum value `SpatialMode.GRID` for `@Spatial` annotation was renamed to `SpatialMode.HASH`.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00112">
+            <when>
+                <javaclass references="org.hibernate.search.event.impl.FullTextIndexEventListener">
+                    <location>INHERITANCE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="FullTextIndexEventListener class is now final in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>The class `FullTextIndexEventListener` is now a final class. If you currently extend this class, you must find an alternate solution to achieve the same functionality. You could in some cases use an `EntityIndexingInterceptor`.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
```

Comment:
Could also link to http://hibernate.org/search/documentation/migrate/5.0/#fulltextindexeventlistener-now-final ?